Mathlib/CategoryTheory/EffectiveEpi/Preserves.lean

```diff
@@ -93,9 +95,126 @@ instance map_effectiveEpi (F : C ⥤ D) [F.PreservesEffectiveEpis] {X Y : C} (f
     [EffectiveEpi f] : EffectiveEpi (F.map f) :=
   PreservesEffectiveEpis.preserves f
 
-instance (F : C ⥤ D) [IsEquivalence F] : F.PreservesEffectiveEpis where
+/--
+A class describing the property of preserving effective epimorphic families.
+-/
+class PreservesEffectiveEpiFamilies (F : C ⥤ D) : Prop where
+  /--
+  A functor preserves effective epimorphic families if it maps effective epimorphic families to
+  effective epimorphic families.
+  -/
+  preserves : ∀ {α : Type*} {B : C} (X : α → C) (π : (a : α) → (X a ⟶ B)) [EffectiveEpiFamily X π],
+    EffectiveEpiFamily (fun a ↦ F.obj (X a)) (fun a  ↦ F.map (π a))
+
+instance map_effectiveEpiFamily (F : C ⥤ D) [F.PreservesEffectiveEpiFamilies]
+    {α : Type*} {B : C} (X : α → C) (π : (a : α) → (X a ⟶ B)) [EffectiveEpiFamily X π] :
+    EffectiveEpiFamily (fun a ↦ F.obj (X a)) (fun a  ↦ F.map (π a)) :=
+  PreservesEffectiveEpiFamilies.preserves X π
+
+/--
+A class describing the property of preserving finite effective epimorphic families.
+-/
+class PreservesFiniteEffectiveEpiFamilies (F : C ⥤ D) : Prop where
+  /--
+  A functor preserves finite effective epimorphic families if it maps finite effective epimorphic
+  families to effective epimorphic families.
+  -/
+  preserves : ∀ {α : Type} [Finite α] {B : C} (X : α → C) (π : (a : α) → (X a ⟶ B))
+    [EffectiveEpiFamily X π],
+    EffectiveEpiFamily (fun a ↦ F.obj (X a)) (fun a  ↦ F.map (π a))
+
+instance map_finite_effectiveEpiFamily (F : C ⥤ D) [F.PreservesFiniteEffectiveEpiFamilies]
+    {α : Type} [Finite α] {B : C} (X : α → C) (π : (a : α) → (X a ⟶ B)) [EffectiveEpiFamily X π] :
+    EffectiveEpiFamily (fun a ↦ F.obj (X a)) (fun a  ↦ F.map (π a)) :=
+  PreservesFiniteEffectiveEpiFamilies.preserves X π
+
+instance (F : C ⥤ D) [PreservesEffectiveEpiFamilies F] : PreservesFiniteEffectiveEpiFamilies F where
+  preserves _ _ := inferInstance
+
+instance (F : C ⥤ D) [PreservesFiniteEffectiveEpiFamilies F] : PreservesEffectiveEpis F where
   preserves _ := inferInstance
 
-end Functor
+instance (F : C ⥤ D) [IsEquivalence F] : F.PreservesEffectiveEpiFamilies where
+  preserves _ _ := inferInstance
 
 end Preserves
+
+section Reflects
+
+/--
+A class describing the property of reflecting effective epimorphisms.
+-/
+class ReflectsEffectiveEpis (F : C ⥤ D) : Prop where
+  /--
+  A functor reflects effective epimorphisms if it only maps effective
+  epimorphisms to effective epimorphisms.
```

Comment:
This docstring doesn't sound quite right... Similarly for the other `reflects` classes.